statsmodels/discrete/tests/test_discrete.py

```diff
@@ -2381,6 +2384,20 @@ def test_unchanging_degrees_of_freedom():
     assert_equal(res3.df_model, res1.df_model)
     assert_equal(res3.df_resid, res1.df_resid)
 
+    check_inherited_attributes(res1)
+    check_inherited_attributes(res2)
+    check_inherited_attributes(res3)
+
+
+def check_inherited_attributes(res):
+    # check that specific attributes are directly defined in the Results
+    # object; this can help ensure that Results objects can still be useful
+    # after `remove_data` and a pickle/unpickle cycle.
+    model = res.model
+    if isinstance(model, MultinomialModel):
+        assert res.J == model.J
+        assert res.K == model.K
```

Comment:
no plain assert in unit test
either numpy assert_ or, preferably, because of better message, assert_equal

(I think I want to avoid pytest messing with and rewriting our test as long as possible. )